pkg/session/session.go

```diff
@@ -2143,7 +2143,11 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	}
 
 	// Execute the physical plan.
-	logStmt(stmt, s)
+	if variable.ProcessDeferGeneralLog.Load() {
+		defer logStmt(stmt, s)
```

Comment:
Clould you add some tests for this case? One way is to mock a new logger for [logutil.GeneralLogger](https://github.com/pingcap/tidb/blob/241c10294b9c0305db590576700c00de306be573/pkg/util/logutil/log.go#L123) and check the log entries in a custom `zapcore.Core`.

Reply:
Okay. Will do.